lua/barbar/ui/render.lua

```diff
@@ -548,6 +549,39 @@ local function get_bufferline_containers(data, bufnrs, refocus)
   return pinned_containers, containers, current_buffer
 end
 
+--- Generate the syntax for the offset on `side`
+--- @param side side
+--- @return string
+local function generate_side_offset(side)
+  local offset = state.offset[side] --- @type barbar.state.offset.side
+
+  local align = offset.align
+  local hl = wrap_hl(offset.hl)
+  local text = offset.text
+  local width = offset.width
+
+  local max_content_width = width - 2
+  local content = nodes.slice_right({ { hl = hl, text = text } }, max_content_width)
+
+  if max_content_width > #text then
+    local offset_nodes = { { hl = hl, text = (' '):rep(width) } } --- @type barbar.ui.node[]
+
+    local insert_position
+    if align == 'left' then
+      insert_position = 1
+    else
+      insert_position = width - #text - 1
+      if align == 'center' then
+        insert_position = ceil(insert_position / 2)
+      end
+    end
```

Comment:
Wouldn't this be more simple to understand as `if align=right... else if align=center ... else`?

Reply:
I agree with you— checking `align`ments one at a time would read easier. My reason for not doing that was since `center_alignment_position == ceil(right_alignment_position / 2)`, it simplified the implementation to allow them to share part of the `else` branch.

Would a comment e.g. `-- else 'right' alignment` make it clearer?

Reply:
I wouldn't mind repeating the line/code if it increases readability tbh. It's a small nit, up to you.

Reply:
I'm personally more concerned about the duplication (in this case anyway— DRY is not _always_ the most important thing). I added comments to clarify the control flow, though :)